### DISCUSSION TITLE
Change Parameters without Reboot?

### DISCUSSION THREAD

**Discussion by eliasgraf** [Louisvdw/dbus-serialbattery#174] [Q&A] [answered]
Is there a possibility to change Parameters/Code without a reboot?
Would it be possible to have a script that reboots only the dbus-serialbattery driver?
best regards!

**Comment by Louisvdw** [comment, marked as answer]
> sh /etc/dbus-serialbattery/restartservice.sh

**Reply by eliasgraf** [reply]
Thanks, curiously havent had this file in my directory... dont know why

**Reply by Louisvdw** [reply]
ah. We only use it in development so it is not part of the release build. You will have to manually copy it from the repo to your GX system
> wget https://github.com/Louisvdw/dbus-serialbattery/blob/master/etc/dbus-serialbattery/restartservice.sh

**Reply by eliasgraf** [reply]
got it!
but when executing i get:
`svc: warning: unable to chdir to /service/dbus-serialbattery: file does not exist`
looking in the /service/ dir, i also cant find it...

**Reply by Louisvdw** [reply]
Is the serialbattery driver already running on your system? The GX manage the /service/ folder and populate it with the devices that is running. If it is not running yet then it will not be there.

**Reply by eliasgraf** [reply]
driver is installed via:
`wget https://raw.githubusercontent.com/Louisvdw/dbus-serialbattery/master/etc/dbus-serialbattery/installrelease.sh`
`sh installrelease.sh`
`reboot`
in /service/ i find:
`root@raspberrypi2:/service# ls`
`can-bus-bms.can0            dbus-generator-starter      dbus-pump                   dbus-valence.can1           mk2-dbus.ttyUSB1            ppp                         vegetty
can-bus-bms.can1            dbus-modbus-client          dbus-qwacs                  dbus-vebus-to-pvinverter    mosquitto                   serial-starter              venus-html5-logger
dbus-adc                    dbus-modbustcp              dbus-serialbattery.ttyUSB0  gui                         mqtt-rpc                    service-advertiser          venus-platform
dbus-ble-sensors            dbus-motordrive.can0        dbus-systemcalc-py          hub4control                 mqtttogrid                  simple-upnpd                vesmart-server
dbus-digitalinputs          dbus-motordrive.can1        dbus-tempsensor-relay       llmnrd                      netmon                      socketcand                  vrmlogger
dbus-fronius                dbus-mqtt                   dbus-valence.can0           localsettings               openssh                     ssh-tunnel                  websockify-c
`
do i have to change to dbus-serialbattery.ttyUSB0 ?

**Reply by Louisvdw** [reply]
your venus makes that mapping to the driver running on port ttyUSB0. You should not manually change anything for that under the /service folder. That command should restart all devices for that driver.

**Reply by eliasgraf** [reply]
hmm... if i change inside restartservice.sh to 
`svc -d -u /service/dbus-serialbattery.ttyUSB0`
the driver obviously restarts because i get a "not connected" inside venus os ui for a short time but unfortunately the new code (i do changes at the CCCM calulations inside battery.py) is not active.
